docs/coverage.md

```
@@ -0,0 +1,217 @@
+
+**Option 1: Use Stable Rust (Current Setup)**
+
+```bash
+# Line and function coverage only
+cargo coverage-text
+cargo coverage  # HTML report with 0% branch coverage
+```
+
+**Result**: Excellent line/function coverage, zeros for branch coverage
+
+**Option 2: Enable Branch Coverage with Nightly**
```

markdownlint-cli2 (0.17.2)

Emphasis used instead of a heading
null

(MD036, no-emphasis-as-heading)